src/Backend/Modules/ContentBlocks/ContentBlock/ContentBlockHandler.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Backend\Modules\ContentBlocks\ContentBlock;
+
+use Backend\Core\Engine\Model;
+use Doctrine\ORM\EntityManager;
+
+class ContentBlockHandler
+{
+    /** @var EntityManager */
+    private $entityManager;
+
+    /**
+     * @param EntityManager $entityManager
+     */
+    public function __construct(EntityManager $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @param CreateContentBlock $createContentBlock
+     *
+     * @return ContentBlock
+     */
+    public function create(CreateContentBlock $createContentBlock)
+    {
+        $contentBlock = ContentBlock::create(
+            $this->entityManager
+                ->getRepository(ContentBlock::class)
+                ->getNextIdForLanguage($createContentBlock->language),
+            $this->getNewExtraId(),
+            $createContentBlock->language,
+            $createContentBlock->title,
+            $createContentBlock->text,
+            !$createContentBlock->isVisible,
+            $createContentBlock->template
+        );
+
+        $this->entityManager->persist($contentBlock);
+        $this->entityManager->flush($contentBlock);
```

Comment:
I had been thinking about that as well. The fork cms event trigger stuff is still required right ?
I was thinking of maybe using a real event here and adding a listener that throws this old event, good idea @WouterSioen ?